File: Examen/Examen_1er_Bimestre/src/com/ejemplo/dao/FileUtil.java
package java.com.ejemplo.dao;

import java.io.*;
import java.util.ArrayList;
import java.util.List;

public class FileUtil {
    // Escribir en un archivo
    public static void writeToFile(String filename, String data) throws IOException {
        writeToFile(filename, data, true);
    }

    // Escribir en un archivo con opción de sobrescribir
    public static void writeToFile(String filename, String data, boolean append) throws IOException {
        try (BufferedWriter writer = new BufferedWriter(new FileWriter(filename, append))) {
            writer.write(data);
            if (append) {
                writer.newLine();
            }
        }
    }

    // Leer de un archivo
    public static List<String> readFromFile(String filename) throws IOException {
        List<String> lines = new ArrayList<>();
        try (BufferedReader reader = new BufferedReader(new FileReader(filename))) {
            String line;
            while ((line = reader.readLine()) != null) {
                lines.add(line);
            }
        }
        return lines;
    }
}

File: Examen_02/src/dao/FileUtil.kt
package dao

import java.io.*


object FileUtil {
    @Throws(IOException::class)
    fun writeToFile(filename: String, data: String, append: Boolean = true) {
        val file = File(filename)
        file.parentFile?.mkdirs()
        BufferedWriter(FileWriter(file, append)).use { writer ->
            writer.write(data)
            if (append) {
                writer.newLine()
            }
        }
    }

    @Throws(IOException::class)
    fun readFromFile(filename: String): List<String> {
        val lines = mutableListOf<String>()
        val file = File(filename)
        if (!file.exists()) {
            return lines
        }
        BufferedReader(FileReader(file)).use { reader ->
            var line = reader.readLine()
            while (line != null) {
                lines.add(line)
                line = reader.readLine()
            }
        }
        return lines
    }
}
